cpp/distance_calculator/distance_calculator.cpp

```diff
@@ -0,0 +1,136 @@
+#include <mgp.hpp>
+#include <math.h>
+# define M_PI 3.14159265358979323846  /* pi */
+# define KM_MUL 0.001
+
+const char *kProcedureSingle = "single";
+const char *kProcedureMultiple = "multiple";
+
+const char *kReturnDistance = "distance";
+const char *kReturnDistances = "distances";
+
+
+const char *kArgumentStartPoint = "start_point";
+const char *kArgumentEndPoint = "end_point";
+const char *kArgumentStart = "start";
+const char *kArgumentEnd = "end";
+
+const char *kArgumentDecimals = "decimals";
+const char *kArgumentMetrics = "metrics";
+
+double distance_calc(const mgp::Node &node1, const mgp::Node &node2, bool use_km, int decimals){
+
+  double lat1 = node1.GetProperty("lat").ValueDouble();
+  double lng1 = node1.GetProperty("lng").ValueDouble();
+  
+  double lat2 = node2.GetProperty("lat").ValueDouble();
+  double lng2 = node2.GetProperty("lng").ValueDouble();
+
+  double pi_rad = M_PI / 180.0;
+  double R = 6371000.0;
+
+  double phi_1 = lat1 * pi_rad;
+  double phi_2 = lat2 * pi_rad;
+
+  double delta_phi = (lat2 - lat1) * pi_rad;
+  double delta_lambda = (lng2 - lng1) * pi_rad;
+
+  double sin_delta_phi = sin(delta_phi / 2);
+  double sin_delta_lambda = sin(delta_lambda /2);
+
+  double a = sin_delta_phi*sin_delta_phi + cos(phi_1)*cos(phi_2)* (sin_delta_lambda*sin_delta_lambda);
```

Comment:
Make sure this file is formatted - whitespace is missing here around `*` and in other lines too.